numpy/core/src/multiarray/buffer.c

```diff
@@ -571,6 +570,7 @@ _buffer_info_new(PyObject *obj)
     return info;
 
 fail:
+    free(fmt.s);
```

Comment:
Hmm, freeing a possibly NULL pointer is supposed to be safe, although there were archaic platforms on which it segfaulted. We'll go with it.